utils/fixed_decimal/src/decimal.rs

```diff
@@ -1663,20 +1661,26 @@ impl FixedDecimal {
             self
         );
 
-        // digits invariants:
-        let max_len = (self.magnitude as i32 - self.lower_magnitude as i32 + 1) as usize;
-        debug_assert!(self.digits.len() <= max_len, "{:?}", self);
-        #[allow(clippy::indexing_slicing)] // TODO(#1668) Clippy exceptions need docs or fixing.
-        if !self.digits.is_empty() {
-            debug_assert_ne!(self.digits[0], 0, "Starts with a zero {:?}", self);
-            debug_assert_ne!(
-                self.digits[self.digits.len() - 1],
-                0,
-                "Ends with a zero {:?}",
+        #[cfg(debug_assertions)]
```

Comment:
Nit: This whole function is already `#[cfg(debug_assertions)]`

Suggestion: Mark the whole function as allow(indexing_slicing) since it is only used in debug mode